Ryujinx.Graphics.Vulkan/PipelineFull.cs

```diff
@@ -0,0 +1,295 @@
+                    {
+                        SType = StructureType.ConditionalRenderingBeginInfoExt,
+                        Buffer = buffer,
+                        Flags = flags
+                    };
+
+                    // Gd.ConditionalRenderingApi.CmdBeginConditionalRendering(CommandBuffer, conditionalRenderingBeginInfo);
+                    return true;
+                }
+            }
+
+            // The GPU will flush the queries to CPU and evaluate the condition there instead.
+
+            FlushCommandsImpl(); // The thread will be stalled manually flushing the counter, so flush GL commands now.
+            return false;
+        }
+
+        public bool TryHostConditionalRendering(ICounterEvent value, ICounterEvent compare, bool isEqual)
+        {
+            FlushCommandsImpl(); // The thread will be stalled manually flushing the counter, so flush GL commands now.
+            return false;
+        }
+
+        public void FlushCommandsImpl([System.Runtime.CompilerServices.CallerMemberName] string caller = "")
+        {
+            // System.Console.WriteLine("flush by " + caller);
+
+            EndRenderPass();
+            PauseTransformFeedbackInternal();
+
+            foreach (var queryPool in _activeQueries)
+            {
+                Gd.Api.CmdEndQuery(CommandBuffer, queryPool, 0);
+            }
+
+            CommandBuffer = (Cbs = Gd.CommandBufferPool.ReturnAndRent(Cbs)).CommandBuffer;
+
+            // Restore per-command buffer state.
+
+            if (Pipeline != null)
+            {
+                Gd.Api.CmdBindPipeline(CommandBuffer, Pbp, Pipeline.Get(Cbs).Value);
+            }
+
+            foreach (var queryPool in _activeQueries)
+            {
+                Gd.Api.CmdResetQueryPool(CommandBuffer, queryPool, 0, 1);
+                Gd.Api.CmdBeginQuery(CommandBuffer, queryPool, 0, 0);
+            }
+
+            ResumeTransformFeedbackInternal();
+            SignalCommandBufferChange();
+        }
+
+        public void BeginQuery(QueryPool pool)
+        {
+            EndRenderPass();
+
+            Gd.Api.CmdResetQueryPool(CommandBuffer, pool, 0, 1);
+            Gd.Api.CmdBeginQuery(CommandBuffer, pool, 0, 0);
+
+            _activeQueries.Add(pool);
+        }
+
+        public void EndQuery(QueryPool pool)
+        {
+            Gd.Api.CmdEndQuery(CommandBuffer, pool, 0);
+
+            _activeQueries.Remove(pool);
+        }
+
+        public void CopyQueryResults(QueryPool pool, BufferHolder holder)
+        {
+            EndRenderPass();
+
+            var buffer = holder.GetBuffer(CommandBuffer, true).Get(Cbs, 0, sizeof(long)).Value;
+
+            Gd.Api.CmdCopyQueryPoolResults(
+                CommandBuffer,
+                pool,
+                0,
+                1,
+                buffer,
+                0,
+                sizeof(long),
+                QueryResultFlags.QueryResult64Bit);
```

Comment:
```suggestion
                QueryResultFlags.QueryResult64Bit | QueryResultFlags.QueryResultWaitBit);
```
The counter queue expects this command to always copy the query result. It seems that this copy can be skipped if the result is not available yet.

This should fix the counter queue getting stuck on AMD. Leischii has reported it fixing counter queue crashes on MK8D, but I'd recommend testing Splatoon 2 to see if the ink works.

Reply:
fwiw I tried this on NVIDIA before, and it didn't fix the splatoon 2 crash.